workflow/controller/controller.go

```diff
@@ -827,7 +830,12 @@ func (wfc *WorkflowController) addWorkflowInformerHandlers(ctx context.Context)
 	wfc.wfInformer.AddEventHandler(
 		cache.FilteringResourceEventHandler{
 			FilterFunc: func(obj interface{}) bool {
-				return reconciliationNeeded(obj.(*unstructured.Unstructured))
+				un, ok := obj.(*unstructured.Unstructured)
+				if !ok {
+					log.Warnf("Workflow FilterFunc: '%v' is not an unstructured", obj)
+					return false
+				}
+				return reconciliationNeeded(un)
```

Comment:
```suggestion
				return ok && reconciliationNeeded(un)
```

Would it be better to keep the same style as here? https://github.com/argoproj/argo-workflows/blob/dd08e95d5b7e77fa7fe4b2230d590c54ec8144f7/workflow/controller/controller.go#L870-L872

Reply:
I guess it depends on whether we think there's a benefit in having the Warn line - do you think there isn't?

Reply:
yep, of course more logs are usually more helpful, I've no objection to this. 
maybe we can think about why other similar lines within informer don't print such logs? 🤔